### PR TITLE
appmanager refactoring and test

### PR DESCRIPTION
**Pull request by rihernandez**

### CODE REVIEW

**Review by psanders — COMMENTED** [review]
@rihernandez @Wardner @speedymonster 

Who is reviewing this?

File: mods/appmanager/src/appmanager.ts [hunk 1]
```diff
@@ -49,12 +60,11 @@ export default class AppManager extends FonosService {
    *
    * @see module:core:FonosService
    */
-  constructor(options?: any) {
+  constructor(options?: unknown) {
```
Comment (speedymonster) [inline mods/appmanager/src/appmanager.ts]: Why didn't you use the **ServiceOptions** interface?

File: mods/appmanager/src/appmanager.ts [hunk 2]
```diff
@@ -79,53 +89,15 @@ export default class AppManager extends FonosService {
...
-
-    if (
-      !fs.existsSync(request.dirPath) ||
-      !fs.lstatSync(request.dirPath).isDirectory()
-    ) {
-      throw new Error(
-        `${request.dirPath} does not exist or is not a directory`
-      );
-    }
+  async deployApp(appPath: string, appRef?: string): Promise<CreateAppManagerRequest> {
```
Comment (speedymonster) [inline mods/appmanager/src/appmanager.ts:93]: appPath should be required because it is not a conditional use variable

**Review by speedymonster — APPROVED** [review]
Slight changes

File: mods/appmanager/src/appmanager.ts [hunk 3]
```diff
@@ -1,18 +1,29 @@
 import Storage from "@fonos/storage";
```
Comment (wardner) [inline mods/appmanager/src/appmanager.ts:20]: Add copywrite notice the header.

File: mods/appmanager/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+export interface AppManagers {
+    ref?: string,
+    name: string,
+    description: string,
+    createTime: string,
+    updateTime: string,
+    status: string,
+    accessKeyId: string,
+    labels: string
+    }
```
Comment (wardner) [inline mods/appmanager/src/types.ts]: This `}` has an additional space. Remove it

File: mods/appmanager/test/appmanager.unit.test.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    const result = await appAPI.listApps(request);
+
+    expect(result.apps[0]).to.have.property("ref").to.be.equal(appObj.getRef());
+    expect(result.apps[0]).to.have.property("name").to.be.equal(appObj.getName());
+    expect(result.apps[0]).to.have.property("description").to.be.equal(appObj.getDescription());
+    expect(result.apps[0]).to.have.property("createTime").not.to.be.null;
+    expect(result.apps[0]).to.have.property("updateTime").not.to.be.null;
+      
+  });
+
```
Comment (wardner) [inline mods/appmanager/test/appmanager.unit.test.ts]: Remove line skip.

File: mods/ctl/src/commands/apps/get.ts [hunk 6]
```diff
@@ -22,13 +22,17 @@ export default class GetCommand extends Command {
...
-        Description: app.getDescription(),
-        Created: moment(app.getCreateTime()).fromNow(),
-        Updated: moment(app.getUpdateTime()).fromNow()
+        Name: app.name,
+        Description: app.description,
+         Created: moment(createTime.toUTCString()).fromNow(),
+         Updated: moment(updateTime.toUTCString()).fromNow()
       };
 
+
```
Comment (wardner) [inline mods/ctl/src/commands/apps/get.ts]: Remove line skip

File: mods/appmanager/src/appmanager.ts [hunk 7]
```diff
@@ -1,18 +1,29 @@
...
-import {FonosService, AppManagerService, AppManagerPB} from "@fonos/core";
-import {App} from "@fonos/core/src/server/protos/appmanager_pb";
-import {View} from "@fonos/core/src/server/protos/common_pb";
+//import {FonosService, AppManagerService, AppManagerPB, ServiceOptions} from "@fonos/core";
+// import {App} from "@fonos/core/src/server/protos/appmanager_pb";
```
Comment (wardner) [inline mods/appmanager/src/appmanager.ts]: Delete those comments

File: mods/appmanager/src/appmanager.ts [hunk 8]
```diff
@@ -1,18 +1,29 @@
...
+
+import {
+  GetAppManagerResponse,
+  ListAppManagerRequest,
+  ListAppManagerResponse,
+  DeleteAppManagerResponse,
+  CreateAppManagerRequest
+} from "./types";
+
+
```
Comment (wardner) [inline mods/appmanager/src/appmanager.ts]: Delete line skip.

**Review by wardner — CHANGES_REQUESTED** [review]
Do the requested changes. 

Reviewed.

File: mods/ctl/src/commands/numbers/create.ts [hunk 9]
```diff
@@ -28,7 +28,7 @@ export default class CreateCommand extends Command {
         pageToken: "1",
         view
       });
-      const apps = res.getAppsList().map((app: AppManagerPB.App) => {
+      const apps = res.apps.map((app: any) => {
```
Comment (psanders) [inline mods/ctl/src/commands/numbers/create.ts:31]: Fix app's type (it's currently "any")

File: mods/ctl/src/commands/numbers/update.ts [hunk 10]
```diff
@@ -27,7 +27,7 @@ export class UpdateCommand extends Command {
       pageToken: "1",
       view
     });
-    const apps = response.getAppsList().map((app: AppManagerPB.App) => {
+    const apps = response.apps.map((app: any) => {
```
Comment (psanders) [inline mods/ctl/src/commands/numbers/update.ts:30]: Fix "any" type

File: mods/ctl/test/commands/apps/deploy.apps.unit.test.ts [hunk 11]
```diff
@@ -18,10 +17,14 @@ describe("@fonos/ctl/apps", () => {
...
-    app.setDescription("A test application");
-    app.setCreateTime("January 01, 1970 00:00:00 UTC.");
+    const app = {
+      name:"My App",
+      description: "A test application",
+      createTime: "January 01, 1970 00:00:00 UTC.",
+      updateTime: "January 01, 1970 00:00:00 UTC.",
+      status: "ACTIVE",
+      accessKeyId: "70lkv20G5MCgUIKYJI6Z"
+  }
```
Comment (psanders) [inline mods/ctl/test/commands/apps/deploy.apps.unit.test.ts:27]: Needs linting

File: mods/storage/src/storage.ts [hunk 12]
```diff
@@ -28,9 +28,11 @@ import {
   getObjectURLResponse,
   UploadObjectResponse
 } from "./types";
-import {promisifyAll} from "grpc-promise";
+
```
Comment (psanders) [inline mods/storage/src/storage.ts:31]: Needs linting

File: mods/storage/src/storage.ts [hunk 13]
```diff
@@ -28,9 +28,11 @@ import {
...
 import {utils} from "./utils";
 
+import {promisifyAll} from "grpc-promise";
+
```
Comment (psanders) [inline mods/storage/src/storage.ts:35]: Needs linting

File: mods/appmanager/src/types.ts [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ *    https://opensource.org/licenses/MIT
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export interface AppManagers {
```
Comment (psanders) [inline mods/appmanager/src/types.ts:19]: This type should simply be `App`. You are not "moving" around and AppManager, just the App,

File: mods/appmanager/src/types.ts [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+  ref?: string;
+  name: string;
+  description: string;
+  createTime: string;
+  updateTime: string;
+  status: string;
+  accessKeyId: string;
+}
+
+export interface CreateAppManagerRequest {
```
Comment (psanders) [inline mods/appmanager/src/types.ts:29]: You are not creating an AppaManager. You simply create an App. `CreateAppRequest` is more appropriate.

Also, this type should not include `createTime`, `updateTime`, or `status.` Those values are created by the backend. The property `description` should be optional.

File: mods/appmanager/src/types.ts [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+  ref?: string;
+  name: string;
+  description: string;
+  createTime: string;
+  updateTime: string;
+  status: string;
+  accessKeyId: string;
+}
+
+export interface CreateAppManagerResponse {
```
Comment (psanders) [inline mods/appmanager/src/types.ts:39]: You are not returning an AppaManager. You simply returning an App. A more appropriate could be `CreateAppResponse`

**Review by psanders — CHANGES_REQUESTED** [review]
The entire `types.ts` file needs to be reviewed. The accessibility to the properties doesn't make sense in most cases. For example, in your type`CreateAppManagerResponse` has createTime and updateTime as optional values, but these values should always be present.

File: mods/ctl/test/commands/apps/deploy.apps.unit.test.ts (hunk 11 above)
Comment (rihernandez) [reply mods/ctl/test/commands/apps/deploy.apps.unit.test.ts:27]: there is not issue apparently in this case. I ran lint and not issue appears.

**Comment by rihernandez** [comment]
deprecated!

**Pull request closed without merging**